CorsixTH/Src/th_map.h

```diff
@@ -564,6 +564,7 @@ class map_scanline_iterator {
   inline int y() const { return y_relative_to_screen; }
   inline const map_tile* get_next_tile() { return tile + tile_step; }
   inline const map_tile* get_previous_tile() { return tile - tile_step; }
+  inline const map_tile* get_adjacent_tile() { return tile + 1; }
```

Comment:
Why did you need this? `tile_step` is +1 or -1 depending on whether it's a forward or backward iterator, so `get_next_tile` or `get_previous_tile` are equivalent and more descriptive.

Reply:
Not true, tile_step is get_width() - 1 or -get_width() + 1, per initialization here https://github.com/CorsixTH/CorsixTH/blob/master/CorsixTH/Src/th_map.cpp#L1781 . forwards has value 2 and backwards 0 from this enum: https://github.com/CorsixTH/CorsixTH/blob/master/CorsixTH/Src/th_map.h#L440

The tile_step steps to x + 1, y - 1 in the fowards direction (i.e. horizontally to the right but at the same vertical height) or to x - 1, y + 1 (i.e. horizontally to the left at the same vertical height).